test/integration/error_output.cc

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <iostream>
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include "sdf/Element.hh"
+#include "sdf/Error.hh"
+#include "sdf/Param.hh"
+#include "sdf/Types.hh"
+#include "test_utils.hh"
+
+////////////////////////////////////////
+// Test Param class for sdf::Errors outputs
+TEST(Error, ErrorOutput)
+{
+  std::stringstream buffer;
+  sdf::testing::RedirectConsoleStream redir(
+    sdf::Console::Instance()->GetMsgStream(), &buffer);
+
+  sdf::Errors errors;
+  ASSERT_NO_THROW(sdf::Param param1("key", "not_valid_type", "true", false,
+                                    errors, "description"));
+  ASSERT_EQ(errors[0].Code(), sdf::ErrorCode::UNKNOWN_PARAMETER_TYPE);
```

Comment:
Sorry, I only meant to use `ASSERT` to check the size of the `errors` object. Here is my suggestion https://github.com/ignitionrobotics/sdformat/pull/1010